prefs.js

```diff
@@ -1,148 +1,130 @@
-    if (GTK_VERSION == 3) {
-      playingSoundNotSupportedLabel.set_line_wrap(true);
-    } else {
+    page.add(group);
+
+    // top bar icon row
+    group.add((() => {
+      const model = new Gio.ListStore({item_type: KeyValuePair});
+      model.splice(0, 0, [
+        new KeyValuePair({key: "when-recording", value: "When recording"}),
+        new KeyValuePair({key: "always", value: "Always"}),
+        new KeyValuePair({key: "never", value: "Never"}),
+      ]);
+      const iconVisibleComboBox = new Adw.ComboRow({
+        title: "Top bar icon",
+        subtitle: "Whether to show top bar icon",
+        model: model,
+        expression: new Gtk.PropertyExpression(KeyValuePair, null, "value"),
+      })
+      for (let i = 0; i < model.n_items; i++) {
+        if (model.get_item(i).key === settings.get_string("icon-visibility", "when-recording")) {
+          iconVisibleComboBox.selected = i;
+          break;
+        }
+      }
+      iconVisibleComboBox.connect("notify::selected-item", () => {
+        const selected_item = iconVisibleComboBox.selected_item;
+        if (selected_item) {
+          settings.set_string("icon-visibility", selected_item.key);
+        }
+      });
+      settings.bind("icon-visibility", iconVisibleComboBox, "active-id", Gio.SettingsBindFlags.DEFAULT);
+      return iconVisibleComboBox;
+    })());
+
+    // osd row
+    const toggleOSD = new Adw.SwitchRow({
+      title: "OSD notification",
+      subtitle: "Whether to show OSD notification",
+    })
+    settings.bind("show-osd", toggleOSD, "active", Gio.SettingsBindFlags.DEFAULT);
+    group.add(toggleOSD);
+
+    // sound notification row
+    const feedbackSoundsSwitch = new Adw.SwitchRow({
+      title: "Sound notification",
+      subtitle: "Play sound when muting and unmuting",
+    })
+    settings.bind("play-feedback-sounds", feedbackSoundsSwitch, "active", Gio.SettingsBindFlags.DEFAULT);
+    group.add(feedbackSoundsSwitch);
+
+    // no sound alert row
+    const isPlayingSoundSupported = Gst != null && GstAudio != null;
+    if (!isPlayingSoundSupported) {
+      const playingSoundNotSupportedLabel = new Gtk.Label({
+        halign: Gtk.Align.START,
+      });
+      playingSoundNotSupportedLabel.set_markup("<span foreground='red'>WARNING. Playing sound is not supported on this system. Is GStreamer package installed?</span>");
       playingSoundNotSupportedLabel.set_wrap(true);
+      group.add(playingSoundNotSupportedLabel);
     }
-    prefsWidget.attach(playingSoundNotSupportedLabel, 0, 5, 1, 1);
-  }
-
-  if (GTK_VERSION == 3) {
-    prefsWidget.show_all();
-  }
-
-  return prefsWidget;
-}
-
-function is_playing_sound_supported() {
-  try {
-    imports.gi.Gst;
-    imports.gi.GstAudio;
-    return true;
-  } catch(e) {
-    log(`${Extension.metadata.uuid}: Playing sound is not supported on this system. Is GStreamer package installed?`);
-    log(`${Extension.metadata.uuid}: ${e}`);
```

Comment:
Slightly late for the review, but @axxapy I wonder, if any warning is printed in the logs or only the message displayed in preferences are shown?